kernel/src/hil/flash.rs

```diff
@@ -12,14 +16,14 @@ pub trait Flash {
     /// when operations complete.
     fn set_client(&self, client: &'static Client);
 
-    /// Read data
-    fn read(&self, offset: usize, buf: &'static mut [u8]);
+    /// Read a page of flash.
+    fn read_page(&self, page_number: usize, buf: &'static mut [u8]) -> ReturnCode;
 
-    /// Write data
-    fn write(&self, offset: usize, buf: &'static mut [u8]);
+    /// Write a page of flash.
+    fn write_page(&self, page_number: usize, buf: &'static mut [u8]) -> ReturnCode;
```

Comment:
Like, the problem with the way it's implemented right now is how the heck do I know I'm passing in something that's not going return an error because it's not exactly the right size? What do I do if it isn't?

I realize the existing HIL doesn't deal with this, but I assume(d) the assumption there was that the implementation would be able to deal with various sizes (e.g. truncate).